File: solutions/ocaml/075-valid-palindrome.ml
(*
  Problem 75: Valid Palindrome (LeetCode 125)
  Difficulty: Easy
  Language: OCaml
*)
let is_palindrome s =
  let cleaned = String.lowercase_ascii s
    |> String.to_seq |> Seq.filter (fun c ->
      (c >= 'a' && c <= 'z') || (c >= '0' && c <= '9'))
    |> List.of_seq in
  cleaned = List.rev cleaned

File: solutions/typescript/075-valid-palindrome.ts
/**
 * Problem 75: Valid Palindrome (LeetCode 125)
 * Difficulty: Easy
 * Language: TypeScript
 */
function isPalindrome(s: string): boolean {
  const cleaned = [...s.toLowerCase()].filter(c => /[a-z0-9]/.test(c));
  const check = (i: number, j: number): boolean =>
    i >= j ? true : cleaned[i] === cleaned[j] && check(i + 1, j - 1);
  return check(0, cleaned.length - 1);
}
